### DISCUSSION TITLE
Wireless Audio Visualizer

### DISCUSSION THREAD

**Discussion by afterBlink** [pschatzmann/arduino-audio-tools#307] [Show and tell]
Completed a project that has an AI-Thinker AudioKit send VU levels and spectrum magnitudes of its analog input to a wireless audio visualizer. Apart from the alternative es8388 library I used, it is entirely based on the arduino-audio-tools library.

[Project description](https://www.dellascolto.com/bitwise/2022/08/14/wireless-audio-visualizer_part_i/)

**Comment by pschatzmann** [comment]
Cool !
Is there any reason why you did not use the AudioKitStream class ? 
It already provides the integration  with the es8388..

**Reply by afterBlink** [reply]
Hi Phil, I started with your streams-audiokit-audiokit example, but that produced heavy distortion on my board. A similar example from thaaraak's es8388 library (that further uses your lib) sounded great, so I proceeded from there. Maybe I could have used the AudioKitStream class as well, but anyway, I still feel that my project owes everything to your library.

**Comment by pschatzmann** [comment]
No worries!
Feel free to file a bug report with the necessary information how to reproduce your issue. It might get tricky however if I can't reproduce it on my boards

**Comment by dizcza** [comment]
Nice authentic-looking project!
I saw a binned equalizer there. But I'm looking for more advanced FFT stuff.
Has anyone implemented FFT spectrum visualization on top of this project?

**Comment by pschatzmann** [comment]
Here is a basic but powerful example project: https://www.pschatzmann.ch/home/2023/02/23/arduino-audiotools-display-fft-result-on-a-led-matrix/ but you can easily adapt the logic to output to any other supported device
